### PR TITLE
HBASE-29009 Depend on jaxws-rt instead jaxws-ri

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by stoty** [comment]
This removes 12 unneccessary JARs but adds duplicate versions in the assembly for two.

I plan to deal with the duplicate dependencies in a separate JIRA.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 18s |  |  master passed  |
| +1 :green_heart: |  compile  |   7m 55s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 56s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 56s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 27s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 27s |  |  The patch does not generate ASF License warnings.  |
|  |   |  43m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Optional Tests | dupname asflicense javac codespell detsecrets xmllint hadoopcheck spotless compile |
| uname | Linux b70256bde4dc 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a7f76173b3644eec976385a47bf81af5992221e2 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 188 (vs. ulimit of 30000) |
| modules | C: hbase-it hbase-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/1/console |
| versions | git=2.34.1 maven=3.9.8 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 41s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 39s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 51s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 18s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 49s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 33s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 51s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 51s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 283m 37s |  |  root in the patch passed.  |
|  |   | 314m 51s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 08c9c4147471 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a7f76173b3644eec976385a47bf81af5992221e2 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/1/testReport/ |
| Max. process+thread count | 8370 (vs. ulimit of 30000) |
| modules | C: hbase-it hbase-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -856,7 +856,7 @@
     <metrics-core.version>3.2.6</metrics-core.version>
     <!--
       Note that the version of jackson-[annotations,core,databind] must be kept in sync with the
-      version of jackson-jaxrs-json-provider shipped in hbase-thirdparty.
+      version of jackson-jax/rs-json-provider shipped in hbase-thirdparty.
```
Comment (Apache9) [inline pom.xml:859]: Why this change? I think 'jackson-jaxrs-json-provider' is the name of the library?

Comment (stoty) [reply pom.xml]: Thanks.
That's just a typo.
Reverted.

**Comment by stoty** [comment]
I'm holding this until we remove jax-ri from the assembly.

After the removal, this will only apply to the test classpath (but simplifying that is also worth doing).

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   2m 28s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 56s |  |  master passed  |
| +1 :green_heart: |  compile  |   7m 41s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 41s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 41s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  xmllint  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  hadoopcheck  |  10m 39s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 41s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 26s |  |  The patch does not generate ASF License warnings.  |
|  |   |  43m 20s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Optional Tests | dupname asflicense javac codespell detsecrets xmllint hadoopcheck spotless compile |
| uname | Linux 591dc38d4c93 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ed88f8ef0e0d3bd9ba47a3a46e08976fe4a25351 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 190 (vs. ulimit of 30000) |
| modules | C: hbase-it hbase-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/2/console |
| versions | git=2.34.1 maven=3.9.8 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   3m 46s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 41s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 52s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 18s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 55s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 55s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 22s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 52s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 306m 59s |  |  root in the patch passed.  |
|  |   | 341m 51s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 81fa0cf60a44 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / ed88f8ef0e0d3bd9ba47a3a46e08976fe4a25351 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/2/testReport/ |
| Max. process+thread count | 9271 (vs. ulimit of 30000) |
| modules | C: hbase-it hbase-assembly . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Review by mnpoonia — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  6s |  |  https://github.com/apache/hbase/pull/6504 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/3/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  6s |  |  https://github.com/apache/hbase/pull/6504 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/6504 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6504/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.
